inventree_kicad/serializers.py

```diff
@@ -519,8 +520,31 @@ def get_description(self, part):
     def annotate_queryset(queryset):
         """Add extra annotations to the queryset."""
 
+        # Annotate with the total variant stock quantity
+        variant_query = variant_stock_query()
         queryset = queryset.annotate(
             in_stock=annotate_total_stock(),
+            allocated_to_sales_orders=annotate_sales_order_allocations(),
+            allocated_to_build_orders=annotate_build_order_allocations(),
+            variant_stock=annotate_variant_quantity(variant_query, reference='quantity')
+        )
+
+        queryset = queryset.annotate(
+            total_in_stock=ExpressionWrapper(
+                F('in_stock') + F('variant_stock'),
+                output_field=DecimalField()
+            )
+        )
+
+        # Annotate with the total 'available stock' quantity
+        # This is the current stock, minus any allocations
+        queryset = queryset.annotate(
+            unallocated_stock=ExpressionWrapper(
+                F('total_in_stock')
+                - F('allocated_to_sales_orders')
+                - F('allocated_to_build_orders'),
+                output_field=DecimalField(),
+            )
```

Comment:
You should wrap this in a `Greatest(..., 0)` to ensure that "unallocated_stock" is never negative.

Ref: https://github.com/inventree/InvenTree/pull/9909